apis/metadata_validation.go

```diff
@@ -60,3 +71,18 @@ func ValidateObjectMetadata(meta metav1.Object) *FieldError {
 
 	return nil
 }
+
+// ValidateCreatorAndModifier validates `metadata.annotation`
+func ValidateCreatorAndModifier(oldSpec, newSpec interface{}, oldAnnotation, newAnnotation map[string]string) *FieldError {
+	var errs *FieldError
+	if oldAnnotation[CreatorAnnotation] != newAnnotation[CreatorAnnotation] {
+		errs = errs.Also(&FieldError{
+			Message: "annotation value is immutable",
+			Paths:   []string{CreatorAnnotation},
+		})
+	}
+	if equality.Semantic.DeepEqual(oldSpec, newSpec) && oldAnnotation[UpdaterAnnotation] != newAnnotation[UpdaterAnnotation] {
```

Comment:
do we care to update, if labels/annotations changed or only spec?
(I think spec only is fine, but just to make sure -- since label/annotation change will trigger a reconcile of child resources).

Reply:
In update case
1. spec change or not change but if `/creator` annotation changes it should give validation error
2. `NO spec` change but only changes to `/lastmodifier` annotation than it should give error
3. Changes to spec and `/lastModifer` will handle automatically and `/lastModifier` get updated if the spec change is done by different user.

So the 2 point will handle
```
if equality.Semantic.DeepEqual(oldSpec, newSpec) && oldAnnotation[UpdaterAnnotation] != newAnnotation[UpdaterAnnotation] 
```